mesonbuild/compilers/compilers.py

```diff
@@ -1288,9 +1288,10 @@ def __init__(self, gcc_type, defines):
                              'b_ndebug', 'b_staticpic',
                              'b_lundef', 'b_asneeded']
 
-    def get_library_dirs(self):
-        env = os.environ.copy()
-        env['LC_ALL'] = 'C'
+    def get_library_dirs(self, env=None):
+        if env is None:
+            env = os.environ.copy()
+            env['LC_ALL'] = 'C'
```

Comment:
This is a totally different `environment`.

Reply:
Agreed. However, in the case we don't have an environment available, we need some fallback scheme.

Reply:
What I mean is, that the `env` passed to `get_library_dirs()` in `compilers/c.py` is an `Environment` object, while this is `os.environ` (a dict). We should not conflate the two. Just rename this to `osenv` or something.

Reply:
Or, better yet, if you change this function to use `_build_wrapper()` like `c.py`, this is no longer a problem.